pkg/daemon/gateway_linux.go

```diff
@@ -379,7 +380,7 @@ func (c *Controller) updateIptablesChain(ipt *iptables.IPTables, table, chain, p
 		added++
 	}
 	for i := len(existingRules) - 1; i >= len(rules)-added; i-- {
-		if err = ipt.Delete(table, chain, strconv.Itoa(i+added)); err != nil {
+		if err = ipt.Delete(table, chain, strconv.Itoa(i+added+1)); err != nil {
```

Comment:
why +1 here

Reply:
This is a bug. I found that when I perform the step of cleaning up the old nat, the last snat rule is never cleaned up. Because this serial number is always 1 smaller than the serial number of iptables